New code of conduct (#1373)
sourcegraph/about: handbook/communication/code_of_conduct.md

# Code of Conduct

## Our pledge

We pledge to make participation in our company, our projects, and our community an open, welcoming, and harassment-free experience for everyone, regardless of ability, age, body size, skin color, culture, education, ethnicity, gender identity and expression, immigration status, level of experience, mental and physical ability, nationality, personal appearance, political beliefs, race, relationship or family status, religion, sex, sexual identity and orientation, or socio-economic status.

We care about the safety and comfort of everyone in our community, including colleagues, contributors, and users.

> 
> 

Reviewer: Watch out for extra whitespace :)

```suggestion

```

## Team member philosophy

> If you work at Sourcegraph or participate in our projects and community, you are important to us. If for any reason you don't feel that,
> [please let us know](#Enforcement).

Reviewer: ```suggestion
If you work at Sourcegraph or participate in our projects and community, you are important to us. If for any reason you don't feel that, [please let us know](#enforcement).
```

Your ultimate success with Sourcegraph will be a function of your personal agency and the environment we create as a team. We want that environment to set you up for success and enable you to thrive while realizing your professional goals and dreams.

Our mission is to enable everyone to unlock the value of code. For us, that means first helping those who code for a living understand the world of code in which they work. Ultimately, it means making the universe of code accessible to billions of people, most of whom currently cannot partake in the code economy. We believe the ability to make a living from code should be available to all.

We live these values within our own organization: We welcome people from a diverse set of backgrounds, and we want every teammate to grow and flourish at Sourcegraph. We believe that a team with a diverse set of viewpoints and experiences united by our common cause will give us the best chance of realizing our mission.

> 
> 

Reviewer: ```suggestion

```

## Our standards

We encourage you to:

> -   **Be welcoming**. Our communities and teams welcome and support people of all backgrounds and identities.
> -   **Be respectful**. We are a worldwide community of professionals, and we conduct ourselves as such. Disagreement is no excuse for poor behavior. Unacceptable behavior includes, but is not limited to:
>     -   Aggressive threats or language.
>     -   Discriminatory or derogatory jokes and language.
>     -   Posting sexually explicit or violent material.
>     -   Trolling.
>     -   Posting, or threatening to post, people's personally identifying information ("doxing").
>     -   Insults, especially those using discriminatory terms or slurs.
>     -   Behavior that could be perceived as unwanted sexual attention or advances.
>     -   Public or private harassment.
>     -   Advocating for or encouraging any of the above behaviors.
> -   **Be friendly and patient**. Remember you might not be communicating in someone else's primary spoken or programming language, and others may not have your level of understanding.
> -   **Understand disagreements**. Disagreements, both social and technical, are useful learning opportunities. Seek to understand other viewpoints, and resolve differences constructively.
> -   **Be comfortable** debating ideas, and keep debates focused on the topic at hand and the concrete actions that will be taken as an outcome of the discussion.
> -   **Be forgiving** in your advice, and careful and precise with the things you produce.
> -   **Exercise empathy.** We all have bad days. Avoid attributing to character flaws what might better be explained by situational factors ([Fundamental Attribution Error](https://en.wikipedia.org/wiki/Fundamental_attribution_error)). Show empathy towards coworkers and community members.
> -   **Praise in front of groups; criticize constructively (1-on-1) when solicited**. Whether you’re an individual contributor, a manager, or a community manager, be clear and concise without language that can be misconstrued as aggressive or personally attacking. Generally, posting unsolicited criticism in public Slack channels is not effective. Praise, however, does belong in public. Some ways we praise at Sourcegraph include:
>     -   Comments in the #thanks Slack channel or in team-specific Slack channels.
>     -   In company or team meetings.
>     -   Sending an eCard/gift card or written note.
>     -   Giving a few hours off (if you’re in a position to do so).
>     -   Sending swag (cup, t-shirt, stickers, socks).
>     -   Sending a tweet or [LinkedIn Kudo](https://blog.linkedin.com/2018/june/12/introducing-linkedin-kudos-say-thanks-and-show-your-appreciation).

Reviewer: ```suggestion
- **Be welcoming**. Our communities and teams welcome and support people of all backgrounds and identities.
- **Be respectful**. We are a worldwide community of professionals, and we conduct ourselves as such. Disagreement is no excuse for poor behavior. Unacceptable behavior includes, but is not limited to:
    - Aggressive threats or language.
    - Discriminatory or derogatory jokes and language.
    - Posting sexually explicit or violent material.
    - Trolling.
    - Posting, or threatening to post, people's personally identifying information ("doxing").
    - Insults, especially those using discriminatory terms or slurs.
    - Behavior that could be perceived as unwanted sexual attention or advances.
    - Public or private harassment.
    - Advocating for or encouraging any of the above behaviors.
- **Be friendly and patient**. Remember you might not be communicating in someone else's primary spoken or programming language, and others may not have your level of understanding.
- **Understand disagreements**. Disagreements, both social and technical, are useful learning opportunities. Seek to understand other viewpoints, and resolve differences constructively.
- **Be comfortable** debating ideas, and keep debates focused on the topic at hand and the concrete actions that will be taken as an outcome of the discussion.
- **Be forgiving** in your advice, and careful and precise with the things you produce.
- **Exercise empathy.** We all have bad days. Avoid attributing to character flaws what might better be explained by situational factors ([Fundamental Attribution Error](https://en.wikipedia.org/wiki/Fundamental_attribution_error)). Show empathy towards coworkers and community members.
- **Praise in front of groups; criticize constructively (1-on-1) when solicited**. Whether you’re an individual contributor, a manager, or a community manager, be clear and concise without language that can be misconstrued as aggressive or personally attacking. Generally, posting unsolicited criticism in public Slack channels is not effective. Praise, however, does belong in public. Some ways we praise at Sourcegraph include:
    - Comments in the #thanks Slack channel or in team-specific Slack channels.
    - In company or team meetings.
    - Sending an eCard/gift card or written note.
    - Giving a few hours off (if you’re in a position to do so).
    - Sending swag (cup, t-shirt, stickers, socks).
    - Sending a tweet or [LinkedIn Kudo](https://blog.linkedin.com/2018/june/12/introducing-linkedin-kudos-say-thanks-and-show-your-appreciation).
```

Keep the following things in mind:

> -   When representing Sourcegraph, be mindful that your words are a reflection on Sourcegraph and the many individuals who make up the team.
>     -   Those in management roles should be especially mindful of their personal beliefs and biases when acting on behalf of the company.
> -   Appreciate uniqueness and do not treat individual opinions as representative of the opinion of the larger group.
> -   Engage others to ensure their voices are heard. Constructive dissent should be addressed directly.
> -   Public polls and private ballots often do not yield the same result. Keep this in mind when gauging consensus.
> -   Handle interpersonal conflict with professionalism and integrity. If there is strong disagreement, feedback should be given directly and clearly, without heat, or implicit denigration.
> -   You will work with others who have personal opinions or beliefs that differ from your own. Strive to overcome these differences and find common cause.
> -   Be aware of power differentials. If you have the ability to threaten someone’s job, your criticism may be taken more harshly than you intend.

Reviewer: ```suggestion
- When representing Sourcegraph, be mindful that your words are a reflection on Sourcegraph and the many individuals who make up the team.
    - Those in management roles should be especially mindful of their personal beliefs and biases when acting on behalf of the company.
- Appreciate uniqueness and do not treat individual opinions as representative of the opinion of the larger group.
- Engage others to ensure their voices are heard. Constructive dissent should be addressed directly.
-  Public polls and private ballots often do not yield the same result. Keep this in mind when gauging consensus.
- Handle interpersonal conflict with professionalism and integrity. If there is strong disagreement, feedback should be given directly and clearly, without heat, or implicit denigration.
- You will work with others who have personal opinions or beliefs that differ from your own. Strive to overcome these differences and find common cause.
- Be aware of power differentials. If you have the ability to threaten someone’s job, your criticism may be taken more harshly than you intend.
```

This code is not exhaustive or complete. It serves to capture our common understanding of a productive, collaborative environment. We expect the code to be followed in spirit as much as in the letter.

> 
> 

Reviewer: ```suggestion

```

## Our responsibilities

> Leadership is responsible for clarifying the standards of acceptable behavior and is expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.

Reviewer: ```suggestion
Leadership is responsible for exemplifying and clarifying the standards of acceptable behavior and is expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.
```

Leadership has the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

> 
> 

Reviewer: ```suggestion

```

## Scope

This code of conduct applies to any individual or community member that is representing the project or its community within both project spaces and public spaces, including (but not limited to) social media, online or offline events, and email or Slack exchanges.

Note: Some Sourcegraph-managed communities have codes of conduct that pre-date this document and issue resolution process. While communities are not required to change their code, they are expected to use the resolution process outlined here. The review team will coordinate with the communities involved to address any concerns.

> 
> 

Reviewer: ```suggestion

```

## Enforcement

We encourage all communities, teams, employees, and team members to resolve issues on their own whenever possible. This builds a broader and deeper understanding and ultimately a healthier interaction.

> Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting conduct@sourcegraph.com. The conduct team is obligated to maintain confidentiality of the person reporting the incident at all times.

Reviewer: ```suggestion
Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting [conduct@sourcegraph.com](mailto:conduct@sourcegraph.com). The conduct team is obligated to maintain the confidentiality of the person reporting the incident at all times.
```

In your report please include:

> -   Your contact information.
> -   Names (real, usernames or pseudonyms) of any individuals involved. If there are additional witnesses, please include them as well.
> -   Your account of what occurred, and if you believe the incident is ongoing. If there is a publicly available record (e.g., a mailing list archive or a public chat log), please include a link or screen captures.
> -   Any additional information that may be helpful.

Reviewer: ```suggestion
- Your contact information.
- Names (real, usernames or pseudonyms) of any individuals involved. If there are additional witnesses, please include them as well.
- Your account of what occurred, and if you believe the incident is ongoing. If there is a publicly available record (e.g., a mailing list archive or a public chat log), please include a link or screen captures.
- Any additional information that may be helpful.
```

All reports will be reviewed by a multi-person team. Where additional perspectives are needed, the team may seek insight from others with relevant expertise or experience. Involved parties are never part of the review team.

Anyone asked to stop unacceptable behavior is expected to comply immediately. If an individual engages in unacceptable behavior, the review team will take appropriate action, including a permanent ban from the community.

> 
> 

Reviewer: ```suggestion

```

# Conflict resolution

_I disagree with the decision to…_

_I disagree with the idea that…_

Disagreements about decisions are a healthy part of team collaboration. By challenging each other individually, our team collectively discovers the best solution.

> 
> 

Reviewer: ```suggestion

```

## Disagreeing with a decision

> When you are challenging a decision, it is important to do so in a constructive way. Failure to do so may lead to [interpersonal conflict](https://about.sourcegraph.com/handbook/communication/conflict_resolution#interpersonal-conflict).

Reviewer: This page has been removed, so we'll need to update the link to reference the section on this page.

```suggestion
When you are challenging a decision, it is important to do so in a constructive way. Failure to do so may lead to [interpersonal conflict](#interpersonal-conflict).
```

Here is a process that you can follow to disagree with a decision productively.

> 1. Proactively try to understand the decision by reviewing available information (Slack conversations, [RFCs](https://about.sourcegraph.com/handbook/communication/rfcs)).

Reviewer: Use path references rather than urls for handbook links.

```suggestion
1. Proactively try to understand the decision by reviewing available information (Slack conversations, [RFCs](rfcs.md)).
```

1. Start a conversation with the decision maker. In this conversation:

>     -   Communicate what information you reviewed in (1) and ask the decision maker to share any information or context that is missing from what you reviewed.
>     -   State clearly that you disagree with the decision and your top 3 reasons for disagreeing. Share why you think these concerns are important or impactful.
>     -   Ask the decision maker how they are thinking about tradeoffs and what they are optimizing for.
>     -   Listen actively.

Reviewer: ```suggestion
    - Communicate what information you reviewed in (1) and ask the decision maker to share any information or context that is missing from what you reviewed.
    - State clearly that you disagree with the decision and your top 3 reasons for disagreeing. Share why you think these concerns are important or impactful.
    - Ask the decision maker how they are thinking about tradeoffs and what they are optimizing for.
    - Listen actively.
```

1. Choose a course of action:

>     -   Agree and commit to the decision.
>     -   Disagree and propose an alternative solution. If your proposal is not accepted, choose a different option.
>     -   Disagree and commit to the decision. Document in an appropriate location (e.g., comment in the relevant Slack thread or relevant RFC) your disagreement, reasons for disagreeing, and the fact that you are committing to the decision.
>     -   Disagree and perform a [clean escalation](https://about.sourcegraph.com/handbook/communication/clean_escalation).

Reviewer: ```suggestion
    - Agree and commit to the decision.
    - Disagree and propose an alternative solution. If your proposal is not accepted, choose a different option.
    - Disagree and commit to the decision. Document in an appropriate location (e.g., comment in the relevant Slack thread or relevant RFC) your disagreement, reasons for disagreeing, and the fact that you are committing to the decision.
    - Disagree and perform a [clean escalation](#clean-escalation).
```

> 
> 

Reviewer: ```suggestion

```

## Interpersonal conflict

_I don’t like working with $NAME because…_

Interpersonal conflict is normal and natural; however, if left unresolved, it can be harmful to the health and productivity of our team. If you are having difficulties working with a teammate (or suspect a teammate is having difficulties working with you), then you should proactively work to resolve the situation by having a direct conversation with the person you are in conflict with. Nonviolent communication<sup><a href="http://www.citizenshandbook.org/nonviolent_communication.pdf">1</a>, <a href="https://www.nonviolentcommunication.com/pdf_files/4part_nvc_process.pdf">2</a></sup> is a useful tool for having productive conversations.

> If you are unable to resolve the conflict directly with the other person, then perform a [clean escalation](https://about.sourcegraph.com/handbook/communication/clean_escalation).

Reviewer: ```suggestion
If you are unable to resolve the conflict directly with the other person, then perform a [clean escalation](#clean-escalation).
```

> 
> 

Reviewer: ```suggestion

```

## Building a shared understanding

> Building a shared understanding is a useful way to prevent and resolve conflicts. It is also a prerequisite for a [clean escalation](https://about.sourcegraph.com/handbook/communication/clean_escalation).

Reviewer: ```suggestion
Building a shared understanding is a useful way to prevent and resolve conflicts. It is also a prerequisite for a [clean escalation](#clean-escalation).
```

You have a shared understanding with another person if:

> 
> 
> 

Reviewer: ```suggestion

```

1. They can describe your point of view accurately.
1. You can describe their point of view accurately.

A shared understanding only requires that you _understand_ the other person’s point of view. You do not need to _agree_ with the other person’s point of view.

Here is a script you can use to build a shared understanding:

> 
> 
> 

Reviewer: ```suggestion

```

1. Alice asks Bob to explain his point of view. > It seems like we are in conflict about X. Could you please explain your point of view to me?
1. Bob explains his point of view.
1. Alice asks any questions that she has about Bob’s point of view so that she can better understand it. Alice does _not_ challenge Bob’s point of view, even if she disagrees with Bob. She is only seeking to understand.
1. Alice says “I heard you say (repeat what Bob said). Is that right?”
1. Bob either says “Yes” (and then the conversation repeats with the roles swapped), or he clarifies his point of view (and then Alice continues the conversation by repeating steps 3 and 4).

If, after establishing a shared understanding, both parties are still unable to move forward, then they should start a clean escalation.

> 
> 

Reviewer: ```suggestion

```

## Clean escalation

Clean escalation is a process that can be used when there is a disagreement on the team about how to best pursue the organizational mission.

The mechanics are simple.

1. Two people have a conflict.

> 1. They attempt to [resolve the conflict](https://about.sourcegraph.com/handbook/communication/conflict_resolution) directly with each other.

Reviewer: ```suggestion
1. They attempt to [resolve the conflict](#conflict-resolution) directly with each other.
```

> 1. If they can’t resolve the conflict, then they come to a [shared understanding](https://about.sourcegraph.com/handbook/communication/conflict_resolution#building-a-shared-understanding) of the situation and agree to cleanly escalate the conflict to their manager(s) _together_.

Reviewer: ```suggestion
1. If they can’t resolve the conflict, then they come to a [shared understanding](#building-a-shared-understanding) of the situation and agree to cleanly escalate the conflict to their manager(s) _together_.
```

1. The manager(s) verify that the pre-conditions for a clean escalation (i.e., the previous two steps) have been met.

>     -   If so, the manager(s) can engage in a conversation to help resolve the conflict.
>     -   If not, the manager(s) do not get involved in the conflict, and instead ask the involved parties to follow this documented clean escalation process.

Reviewer: ```suggestion
    - If so, the manager(s) can engage in a conversation to help resolve the conflict.
    - If not, the manager(s) do not get involved in the conflict, and instead ask the involved parties to follow this documented clean escalation process.
```

1. If the manager(s) are not able to resolve the conflict, they can cleanly escalate to the next level of management. This process continues until the issue is resolved.

> 
> 

Reviewer: ```suggestion

```

# Code of Conduct

> [TOC]

Reviewer: I don't think we need a TOC here, it will be automatically generated and shown on the right of the page based on headings.

Author: Gotcha, fixed

> 

Reviewer: Extra whitespace.

```suggestion
```

## Our pledge

We pledge to make participation in our company, our projects, and our community an open, welcoming, and harassment-free experience for everyone, regardless of ability, age, body size, skin color, culture, education, ethnicity, gender identity and expression, immigration status, level of experience, mental and physical ability, nationality, personal appearance, political beliefs, race, relationship or family status, religion, sex, sexual identity and orientation, or socio-economic status.

We care about the safety and comfort of everyone in our community, including colleagues, contributors, and users.

## Team member philosophy

If you work at Sourcegraph or participate in our projects and community, you are important to us. If for any reason you don't feel that,

> 
> <p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: undefined internal link (link text: "please let us know"). Did you generate a TOC? </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
> 

Reviewer: I think this accidentally made it in from the docs to markdown conversion.

```suggestion
```

Author: It did, ugh. Good looking out

> [please let us know](#heading=h.8o2hfb3bhng).

Reviewer: What should this link to? Looks like this is linking to the gdoc heading.

Author: Yeah, it's supposed to link to the Enforcement heading. I'll fix it.

Your ultimate success with Sourcegraph will be a function of your personal agency and the environment we create as a team. We want that environment to set you up for success and enable you to thrive while realizing your professional goals and dreams.

Our mission is to enable everyone to unlock the value of code. For us, that means first helping those who code for a living understand the world of code in which they work. Ultimately, it means making the universe of code accessible to billions of people, most of whom currently cannot partake in the code economy. We believe the ability to make a living from code should be available to all.

We live these values within our own organization: We welcome people from a diverse set of backgrounds, and we want every teammate to grow and flourish at Sourcegraph. We believe that a team with a diverse set of viewpoints and experiences united by our common cause will give us the best chance of realizing our mission.

> ## **Our standards**

Reviewer: Headings don't need bold styling.

```suggestion
## Our standards
```

We encourage you to:

> *   **Be welcoming**. Our communities and teams welcome and support people of all backgrounds and identities.

Reviewer: I know it's more work, but I'd love to see clean markdown - perhaps we need a Sourcegraph markdown styleguide :) ? I think the preference is to us - instead of * and make sure there is only 1 space between - and the list item.

Author: Okie doke, fixed

I don’t like working with $NAME because…

Interpersonal conflict is normal and natural; however, if left unresolved, it can be harmful to the health and productivity of our team. If you are having difficulties working with a teammate (or suspect a teammate is having difficulties working with you), then you should proactively work to resolve the situation by having a direct conversation with the person you are in conflict with. Nonviolent communication<sup><a href="http://www.citizenshandbook.org/nonviolent_communication.pdf">1</a>, <a href="https://www.nonviolentcommunication.com/pdf_files/4part_nvc_process.pdf">2</a></sup> is a useful tool for having productive conversations.

If you are unable to resolve the conflict directly with the other person, then perform a [clean escalation](https://about.sourcegraph.com/handbook/communication/clean_escalation).

## **Building a shared understanding**

Building a shared understanding is a useful way to prevent and resolve conflicts. It is also a prerequisite for a [clean escalation](https://about.sourcegraph.com/handbook/communication/clean_escalation).

You have a shared understanding with another person if:

> 1. They can describe your point of view accurately.
> 2. You can describe their point of view accurately.

Reviewer: Numbered lists should use markdown styles (do not need to increment, it will happen automatically with repeated 1.)

```suggestion
1. They can describe your point of view accurately.
1. You can describe their point of view accurately.
```
